### PR TITLE
DR2-1461 Move TNA to Preservica role.

### PR DESCRIPTION
**Pull request by MancunianSam**
To create the DataSync locations and tasks, the role which creates it needs to be able to access the Preservica bucket. This means that terraform needs to be able to assume the role which means it needs to already exist in dr2-terraform-backend.

It’s probably safer to move the resources from the terraform environments state to the terraform backend state rather than delete and recreate in case we break something.

This role is a bit of a weird hybrid but it's working.

### CODE REVIEW

File: templates/iam_role/tna_to_preservica_trust_policy.json.tpl [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "AWS": "arn:aws:iam::${management_account_id}:root"
```
Comment (kurtismash) [inline templates/iam_role/tna_to_preservica_trust_policy.json.tpl:7]: Is this so Terraform can make the DataSync location? Can we lock this down any more?

File: templates/iam_role/tna_to_preservica_trust_policy.json.tpl [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      },
+      "Action": "sts:AssumeRole"
+    },
+    {
+      "Sid": "AllowStepFunctionRole",
+      "Effect": "Allow",
+      "Principal": {
+        "AWS": "arn:aws:iam::${account_id}:role/${title_environment}-ingest-role"
```
Comment (kurtismash) [inline templates/iam_role/tna_to_preservica_trust_policy.json.tpl:15]: Why is this needed?

**Review by kurtismash — COMMENTED** [review]
See inline comments

**Comment by kurtismash** [comment]
+ Do we need to add a process to the runbook for user access to Preservica's buckets?

File: templates/iam_policy/tna_to_preservica_copy.json.tpl [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      "Resource": [
+        "arn:aws:s3:::com.preservica.${preservica_tenant}.bulk1",
+        "arn:aws:s3:::com.preservica.${preservica_tenant}.bulk2",
+        "arn:aws:s3:::com.preservica.${preservica_tenant}.bulk3",
+        "arn:aws:s3:::com.preservica.${preservica_tenant}.bulk4",
+        "arn:aws:s3:::com.preservica.${preservica_tenant}.put.holding",
+        "arn:aws:s3:::${ingest_staging_cache_bucket_name}"
+      ],
+      "Sid": "listBucketPreservica"
+    }
```
[the remark this answers is not in the document]
Comment (MancunianSam) [reply templates/iam_policy/tna_to_preservica_copy.json.tpl:77]: So I did this and it stopped working. It started saying it didn't have permission to PutObject. I don't know why so I put it back.

File: templates/iam_role/tna_to_preservica_trust_policy.json.tpl (hunk 1 above)
Comment (MancunianSam) [reply templates/iam_role/tna_to_preservica_trust_policy.json.tpl:7]: It is yeah but it's not the terraform role that's assuming the role, it's the role of whoever is running it. So for the most part, that's the GitHub actions role but I've added in the management admin role as well so we can run `terraform plan` (but never `terraform apply`) locally.

**Review by kurtismash — DISMISSED** [review]

**Review by kurtismash — APPROVED** [review]
